src/main/java/com/group12/ciserver/controller/EventController.java

```diff
@@ -16,4 +24,37 @@
         log.info("Received push event, pushEvent={}", pushEvent);
         return ResponseEntity.noContent().build();
     }
+
+    @GetMapping("/history")
+    @ResponseBody
+    public ArrayList<BuildInfo> buildHistory() {
+
+        DatabaseWrapper db = new DatabaseWrapper();
+
+        return db.getAllBuilds();
+    }
+
+    @GetMapping("/history/{buildId}")
+    @ResponseBody
+    public ResponseEntity<BuildInfo> buildInfo(@PathVariable String buildId) {
+        long uid;
+        try {
+            uid = Integer.parseInt(buildId);
+        } catch (NumberFormatException e) {
+            System.err.println("Bad format for build ID: " + e.getMessage());
+            return ResponseEntity.badRequest().build();
+        }
+
+        DatabaseWrapper db = new DatabaseWrapper();
+        // Query database for the buildId
+        BuildInfo b = db.getBuildInfo(uid);
+
+        if (b != null)
+        {
+            return ResponseEntity.ok(b);
+        } else {
+            return ResponseEntity.noContent().build();
```

Comment:
I would say that a http status 404 (not found) is more appropriate then 204 (no content). Consider using not found instead.

Reply:
sure